cea/analysis/senstivity_demand.py

```diff
@@ -0,0 +1,192 @@
+                results_1.append(sobol_result['S1'])
+                results_2.append(sobol_result['ST'])
+                results_3.append(sobol_result['ST_conf'])
+            else:
+                morris_result = morris.analyze(problem, samples, simulations_parameter, grid_jump=grid, num_levels=levels)
+                results_1.append(morris_result['mu_star'])
+                results_2.append(morris_result['sigma'])
+                results_3.append(morris_result['mu_star_conf'])
+        pd.DataFrame(results_1, columns = problem['names']).to_excel(writer,parameter+'S1')
+        pd.DataFrame(results_2, columns=problem['names']).to_excel(writer, parameter+'ST')
+        pd.DataFrame(results_3, columns=problem['names']).to_excel(writer, parameter + 'conf')
+
+    writer.save()
+    gv.log('Sensitivity Sobol method done - time elapsed: %(time_elapsed).2f seconds', time_elapsed=time.clock() - t0)
+
+# function to call cea
+def screening_cea_multiprocessing(samples, names, output_parameters, locator, weather_path, gv):
+    """
+    This functions calls the simulation of samples in the CEA using multiprocessing
+
+    :param samples: list of lists of samples to simulate
+    :param names: names of buildings to simulate
+    :param output_parameters: list of output parameters to analyse
+    :param locator: path to locator class
+    :param weather_path:  path to weather file
+    :param gv: global variables class
+    :return: List of Dataframes with output of simulation. It relates the results
+    of every outputparameter per building simulated.
+
+    """
+    pool = mp.Pool()
+    gv.log("Using %i CPU's" % mp.cpu_count())
+    joblist = [pool.apply_async(screening_cea, [counter, sample,  names, output_parameters, locator, weather_path, gv])
+               for sample, counter in zip(samples, range(len(samples)))]
+    results = [job.get() for job in joblist]
+    # return in order
+    results = sorted(results, key=lambda tup: tup[0])
+    results = [x[1] for x in results]
+    return results
+
+def screening_cea(counter, sample,  var_names, output_parameters, locator, weather_path, gv):
+    """
+
+    :param counter: counter keepin gth enumber of samples analyzed. it serves to sort the results from asynchronous
+    multiprocessing in order
+    :param sample: sample to simulate
+    :param var_names: names of variables to analyze
+    :param output_parameters: list of output parameters to analyse
+    :param locator: path to locator class
+    :param weather_path: path to weather file
+    :param gv: global variables class
+    :return: Dataframes with output of simulation of a building. It relates the results
+    of every outputparameter.
+    """
+
+    #create a dict with the new input vatiables form the sample and pass in gv
+    gv.samples = dict(zip(var_names, sample))
+    result = None
+    while result is None:  # trick to avoid that arcgis stops calculating the days and tries again.
```

Comment:
Well i beleive it is kind of necessary now as i noticed the tool fails several times. what iam trying to do is to keep it running and when there is an error show me which sample made the error and where it happned. I guess the sensitivity analysis also works to see the extreme cases were the tool fails.